### ISSUE TITLE
Prefetching the same query and variables hangs forever.

### ISSUE THREAD

**Issue by epiqueras** [relay-tools/relay-hooks#115]
I have this code that runs on route changes:

```js
connectToRouteChange((path, query) => {
  if (queries[path]) {
    prefetch.next(environment, queries[path], query);
     // setTimeout(() => prefetch.getValue().retry(), 0);
  }
})
```

Without that commented line, navigating to a `path + query` that was already visited hangs forever.

**Comment by morrys** [comment]
hi @epiqueras,
can you detail the problem better?

What do you mean by `hangs forever`?

the default fetchPolicy is store-or-network (reuse data cached in the store; if the whole query is cached, skip the network request)

**Comment by epiqueras** [comment]
@morrys it just never loads the next query.

**Comment by morrys** [comment]
Are you using it with the usePreloadedQuery hook?

**Comment by epiqueras** [comment]
Yes

**Comment by epiqueras** [comment]
Here is the code: https://github.com/Proof-Of-Humanity/proof-of-humanity-web/blob/master/components/relay-provider.js.

**Comment by morrys** [comment]
hi @epiqueras,
in this PR https://github.com/relay-tools/relay-hooks/pull/116 I think I have recreated the test case of your problem and its resolution.

Before I can officially release it I have to do some more in-depth tests :)

In the meantime, could you try this version and check that your problem is solved?

Thanks

**Comment by epiqueras** [comment]
https://github.com/relay-tools/relay-hooks/pull/116#issuecomment-698612195

>Yes, this does the trick, thank you!
>
>So the issue was that it was querying using a disposed of query fetcher?

**Comment by morrys** [comment]
hi @epiqueras, the fix is released with version 3.7.0.
https://github.com/relay-tools/relay-hooks/releases/tag/v3.7.0

Let me know if you run into any other problems :)

**Comment by epiqueras** [comment]
Updated! Works great 😄 Thank you!
